test/src/test-capi-dense-array-dimension-label.cc

```diff
@@ -0,0 +1,401 @@
+    "Round trip dimension label data for dense 1d array",
+    "[capi][query][DimensionLabel]") {
+  // Vectors for input data.
+  std::vector<double> input_label_data{};
+  std::vector<double> input_attr_data{};
+
+  // Vectors for expected output data.
+  std::vector<double> label_data_sorted_by_label{};
+  std::vector<uint64_t> index_data_sorted_by_label{};
+
+  // Dimension label parameters.
+  tiledb_label_order_t label_order;
+
+  SECTION("Write increasing labels and array", "[IncreasingLabels]") {
+    // Set the label order.
+    label_order = TILEDB_INCREASING_LABELS;
+
+    // Set the data values.
+    input_label_data = {-1.0, 0.0, 0.5, 1.0};
+    input_attr_data = {0.5, 1.0, 1.5, 2.0};
+
+    // Define expected output data.
+    label_data_sorted_by_label = input_label_data;
+    index_data_sorted_by_label = {0, 1, 2, 3};
+  }
+
+  SECTION("Write increasing labels only", "[IncreasingLabels]") {
+    // Set the label order.
+    label_order = TILEDB_INCREASING_LABELS;
+
+    // Set the data values.
+    input_label_data = {-1.0, 0.0, 0.5, 1.0};
+
+    // Define expected output data.
+    label_data_sorted_by_label = input_label_data;
+    index_data_sorted_by_label = {0, 1, 2, 3};
+  }
+
+  SECTION("Write decreasing labels", "[DecreasingLabels]") {
+    // Set the label order.
+    label_order = TILEDB_DECREASING_LABELS;
+
+    // Set the data values.
+    input_label_data = {1.0, 0.0, -0.5, -1.0};
+    input_attr_data = {0.5, 1.0, 1.5, 2.0};
+
+    // Define expected output data.
+    label_data_sorted_by_label = {-1.0, -0.5, 0.0, 1.0};
+    index_data_sorted_by_label = {3, 2, 1, 0};
+  }
+
+  SECTION("Write decreasing labels only", "[DecreasingLabels]") {
+    // Set the label order.
+    label_order = TILEDB_DECREASING_LABELS;
+
+    // Set the data values.
+    input_label_data = {1.0, 0.0, -0.5, -1.0};
+
+    // Define expected output data.
+    label_data_sorted_by_label = {-1.0, -0.5, 0.0, 1.0};
+    index_data_sorted_by_label = {3, 2, 1, 0};
+  }
+
+  SECTION("Write unordered labels and array", "[UnorderedLabels]") {
+    // Set the label order.
+    label_order = TILEDB_UNORDERED_LABELS;
+
+    // Set the data values.
+    input_label_data = {-0.5, 1.0, 0.0, -1.0};
+    input_attr_data = {0.5, 1.0, 1.5, 2.0};
+
+    // Define expected output data.
+    label_data_sorted_by_label = {-1.0, -0.5, 0.0, 1.0};
+    index_data_sorted_by_label = {3, 0, 2, 1};
+  }
+
+  SECTION("Write unordered labels only", "[UnorderedLabels]") {
+    // Set the label order.
+    label_order = TILEDB_UNORDERED_LABELS;
+
+    // Set the data values.
+    input_label_data = {-0.5, 1.0, 0.0, -1.0};
+
+    // Define expected output data.
+    label_data_sorted_by_label = {-1.0, -0.5, 0.0, 1.0};
+    index_data_sorted_by_label = {3, 0, 2, 1};
+  }
+
+  // Create and write the array.
+  create_example(label_order);
+  write_array_with_label(input_attr_data, input_label_data);
+
+  // Check the dimension label arrays have the correct data.
```

Comment:
They provide the scope on the INFO so it will only be printed if there is a failure in the relevant test (see [Catch2 docs](https://github.com/catchorg/Catch2/blob/devel/docs/logging.md#logging-macros)). It was really hard to tell which tests were failing otherwise.